src/ColinHDev/CPlot/provider/DataProvider.php

```diff
@@ -1299,4 +1312,222 @@ public function savePlotRate(Plot $plot, PlotRate $plotRate) : Generator {
     public function close() : void {
         $this->database->close();
     }
+
+    /**
+     * @phpstan-return Generator<mixed, mixed, mixed, void>
+     */
+    private function importData() : Generator {
```

Comment:
I am not really a fan of how importing the world data in the `MyPlotConversionListener` and importing the plot data in this method are split although they are connected.
If you agree, the method could be changed to accept a `World` parameter from which it imports all the data.

Reply:
This will make it difficult to tell when all data imports have completed. The current system renames the old MyPlot config to prevent data loss while being able to tell the data has already imported. That method won't be possible if the function is called for every loaded world.